deebot_client/util/enum.py

```diff
@@ -0,0 +1,52 @@
+"""Enum util."""
+
+from __future__ import annotations
+
+from enum import IntEnum, StrEnum
+from typing import Self
+
+
+class StrEnumWithXml(StrEnum):
+    """String enum with xml value."""
+
+    xml_value: str
+
+    def __new__(cls, value: str, xml_value: str = "") -> Self:
+        """Create new StrEnumWithXml."""
+        obj = str.__new__(cls, value)
+        obj._value_ = value
+        obj.xml_value = xml_value
+        return obj
+
+    @classmethod
+    def from_xml(cls, value: str) -> Self:
+        """Convert from xml value."""
+        for member in cls:
+            if member.xml_value == value:
+                return member
```

Comment:
```suggestion
        if value:
            for member in cls:
                if member.xml_value == value:
                    return member
```

To prevent empty or None values